### PR TITLE
[DOC] Enhanced RDoc for String

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #split
- #each_line
- #lines
- #each_byte
- #bytes

### CODE REVIEW

File: doc/string.rdoc
Comment (peterzhu2118) [inline doc/string.rdoc:54]: I feel it's kind of redundant to say `8-bit bytes` since basically all modern systems have `1 byte == 8 bits`. If the system has `1 byte != 8 bits` I think this would return each byte (and not each 8 bits).

File: doc/string.rdoc
Comment (peterzhu2118) [inline doc/string.rdoc:61]: I should probably have brought this up earlier, but moving the documentation out of the source (`string.c`) makes the `Toggle source` feature of RDoc useless. I think we should find a way to keep that feature functional. I see @duerst brought this up many years ago in ruby/rdoc#409. Has there been any progress on that issue?

Comment (BurdetteLamar) [reply doc/string.rdoc:61]: > I should probably have brought this up earlier, but moving the documentation out of the source (`string.c`) makes the `Toggle source` feature of RDoc useless. I think we should find a way to keep that feature functional. I see @duerst brought this up many years ago in [ruby/rdoc#409](https://github.com/ruby/rdoc/issues/409). Has there been any progress on that issue?

I think it's important to be able to show non-US-ASCII in the examples, especially for classes such as String and Encoding.  I'm thinking that *many* methods's documentation, in String, Encoding, and perhaps others, would profit from examples outside of US-ASCII.

So the issue, as I see it, is whether such examples should wait for an RDoc resolution, or we should go ahead and put them into the C code as we go.  Does this need to go to the developers' meeting?

File: doc/string.rdoc
Comment (BurdetteLamar) [reply doc/string.rdoc]: Done.

File: doc/string.rdoc
Comment (peterzhu2118) [reply doc/string.rdoc:61]: I absolutely agree that we should have non-ASCII examples. We should think of a solution that doesn't break any of the existing features of RDoc. I won't let this issue block you, just something for us to think about in parallel.

File: doc/string.rdoc
Comment (peterzhu2118) [inline doc/string.rdoc:142]: `line_sep` is the empty string, I think you mean newlines

```suggestion
  #  of two or more newlines:
```

File: doc/string.rdoc
Comment (peterzhu2118) [inline doc/string.rdoc:92]: I think there's an extra empty line here.

File: doc/string.rdoc
Comment (BurdetteLamar) [reply doc/string.rdoc:61]: @peterzhu2118, this is ready, then?


After this merge, I'll make include files for a couple of String methods; then we can get some comments from other committers?

File: doc/string.rdoc
Comment (peterzhu2118) [inline doc/string.rdoc:183]: I think there's a typo here

```suggestion
  #  the split occurs at each occurrence of +field_sep+;
```

File: doc/string.rdoc
Comment (peterzhu2118) [reply doc/string.rdoc:61]: I just put in a few more comments

File: doc/string.rdoc
Comment (BurdetteLamar) [reply doc/string.rdoc]: Fixed.

File: doc/string.rdoc
Comment (BurdetteLamar) [reply doc/string.rdoc]: Fixed.

File: doc/string.rdoc
Comment (BurdetteLamar) [reply doc/string.rdoc]: Fixed.

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**
